File: Reservation/src/main/java/net/travel/reservation/dto/PaiementRequestDTO.java
package net.travel.reservation.dto;

import lombok.Data;
import net.travel.reservation.entites.MethodePaiement;
import net.travel.reservation.entites.TypePaiement;

import java.math.BigDecimal;
import java.time.LocalDate;

@Data
public class PaiementRequestDTO {
    private BigDecimal montant;
    private LocalDate datePaiement;
    private TypePaiement typePaiement;
    private MethodePaiement methodePaiement;
    private Long reservationId;
    private String notes;
}

File: reservation-frontend/src/app/core/models/PaiementRequestDTO.ts
import { MethodePaiement } from "./paiement/MethodePaiement.enum";
import { TypePaiement } from "./paiement/TypePaiement.enum";


export interface PaiementRequestDTO {
  montant: number;
  datePaiement: string;
  typePaiement: TypePaiement;
  methodePaiement: MethodePaiement;
  reservationId: number;
  notes?: string;
}